services/data/src/links/RestAPILink/queryToRequestOptions/requestContentType.ts

```diff
@@ -0,0 +1,101 @@
+import { ResolvedResourceQuery, FetchType } from '../../../engine'
+import * as textPlainMatchers from './textPlainMatchers'
+import * as multipartFormDataMatchers from './multipartFormDataMatchers'
+
+type RequestContentType =
+    | 'application/json'
+    | 'text/plain'
+    | 'multipart/form-data'
+    | null
+
+type FormData = {
+    [key: string]: string | Blob
+}
+
+const resourceExpectsTextPlain = (
+    type: FetchType,
+    query: ResolvedResourceQuery
+) =>
+    Object.values(textPlainMatchers).some(textPlainMatcher =>
+        textPlainMatcher(type, query)
+    )
+
+const resourceExpectsMultipartFormData = (
+    type: FetchType,
+    query: ResolvedResourceQuery
+) =>
+    Object.values(multipartFormDataMatchers).some(multipartFormDataMatcher =>
+        multipartFormDataMatcher(type, query)
+    )
+
+export const FORM_DATA_ERROR_MSG =
+    'Could not convert data to FormData: object does not have own enumerable string-keyed properties'
+
+const convertToFormData = (data: FormData) => {
```

Comment:
Shouldn't the type of the `data` input be a Javascript Object, not FormData here?  The output type of the function should be FormData, I believe:

```
const convertToFormData = (data: Record<string, any>): FormData => {
  // ...
}
```

(untested) note this assumes the input will always be an object, not a primitive js value like string, number, etc - if we want to handle those cases we could make the input type `any` and handle the other cases gracefully in this function